shotover-proxy/examples/windsock/cassandra_protocols.rs

```diff
@@ -0,0 +1,219 @@
+            ),
+            (
+                "shotover".to_owned(),
+                match self.shotover {
+                    Shotover::None => "none".to_owned(),
+                    Shotover::Standard => "standard".to_owned(),
+                    _ => todo!(),
+                },
+            ),
+        ]
+        .iter()
+        .cloned()
+        .collect()
+    }
+
+    async fn run(
+        &self,
+        flamegraph: bool,
+        _local: bool,
+        runtime_seconds: u32,
+        operations_per_second: Option<u64>,
+        reporter: UnboundedSender<Report>,
+    ) {
+        let address = match (&self.topology, &self.shotover) {
+            (Topology::Single, Shotover::None) => "127.0.0.1:9043",
+            (Topology::Single, Shotover::Standard) => "127.0.0.1:9042",
+            (Topology::Cluster3, Shotover::None) => "172.16.1.2:9044",
+            (Topology::Cluster3, Shotover::Standard) => "127.0.0.1:9042",
+            (_, _) => todo!(),
+        };
+
+        let config_dir = match &self.topology {
+            Topology::Single => "tests/test-configs/cassandra-passthrough",
+            Topology::Cluster3 => "tests/test-configs/cassandra-cluster-v4",
+        };
+
+        {
+            let _db_instance = docker_compose(&format!("{config_dir}/docker-compose.yaml"));
+            let shotover = match self.shotover {
+                Shotover::Standard => Some(
+                    ShotoverProcessBuilder::new_with_topology(&format!(
+                        "{config_dir}/topology.yaml"
+                    ))
+                    .start()
+                    .await,
+                ),
+                Shotover::None => None,
+                _ => todo!(),
+            };
+            let perf = if flamegraph {
+                if let Some(shotover) = &shotover {
+                    Some(Perf::new(shotover.child.as_ref().unwrap().id().unwrap()))
+                } else {
+                    todo!()
+                }
+            } else {
+                None
+            };
+
+            let cluster_config = NodeTcpConfigBuilder::new()
+                .with_contact_point(address.into())
+                .with_version(match self.protocol {
+                    CassandraProtocol::V3 => Version::V3,
+                    CassandraProtocol::V4 => Version::V4,
+                    CassandraProtocol::V5 => Version::V5,
+                })
+                .build()
+                .await
+                .unwrap();
+            let session = Arc::new(
+                TcpSessionBuilder::new(RoundRobinLoadBalancingStrategy::new(), cluster_config)
+                    .build()
+                    .await
+                    .unwrap(),
+            );
+
+            let row_count = 1000usize;
+
+            session.query("CREATE KEYSPACE IF NOT EXISTS ks WITH REPLICATION = { 'class' : 'SimpleStrategy', 'replication_factor' : 1 }").await.unwrap();
+            // session.await_schema_agreement().await.unwrap();
```

Comment:
do we need to replace these with a cdrs equivalent or does cdrs do this automatically?

Reply:
I'm not sure if it does it automatically but waiting for one second seems to work, I'd imagine it receives the keyspace change event in time.